packages/nimble-components/src/wafer-map/specs/features/rendering.md

```diff
@@ -82,7 +82,49 @@ The POC is found in this branch [Worker Rendering POC](https://github.com/ni/nim
+3. The transfer method for arrow tables is cumbersome, we would have to use another higher level library [geoarrow](https://github.com/geoarrow/geoarrow-js/blob/main/src/worker/transferable.ts). Fortunately we can skip over this problem by not transferring the tables.
+4. The iteration over stored rows is very bad compared to typed arrays as seen in the table below. This impacts the goals we set for this rendering improvement. The solution to this issue and the transferring issue is splitting the relevant columns from the table (rows, columns, values, tags mask) and messaging them to the worker separately. This can be done with a very small overhead using the [getChild](https://arrow.apache.org/docs/js/classes/Arrow_dom.Table.html#getChild) method and calling [toArray](https://arrow.apache.org/docs/js/classes/Arrow_dom.Vector.html#toArray) on the resulting vector. After being transferred, The buffers can be cached to speed up value access and filtering.
+
+| name                      | duration (ms) [1]  | duration (ms) [2]  | detail                                                                                                    |
+| ------------------------- | ------------------ | ------------------ | --------------------------------------------------------------------------------------------------------- |
+| typed iterate             | 7.551699995994568  | 6.052600026130676  | iterating over two 1M typed arrays and calculating the sums                                               |
+| typed from table iterate  | 6.4953999519348145 | 5.136900067329407  | iterating over two 1M typed arrays from Table columns and calculating the sums (time includes conversion) |
+| vector iterate            | 76.4708000421524   | 66.58230006694794  | iterating over two 1M Vectors and calculating the sums                                                    |
+| table get() iterate       | 1350.0404000282288 | 1030.582899928093  | iterating over the 1M Table using `table.get(rowIndex)` and calculating the sums                          |
+| table [iterator] iterate  | 1091.6706000566483 | 1011.069100022316  | iterating over the 1M Table using the [iterator] and calculating the sums                                 |
+| array from table iterate  | 943.0076999664307  | 980.0875999927521  | iterating over the 1M Table after converting `toArray()` and calculating the sums                         |
+| vector from table iterate | 965.2465000152588  | 1012.9023000001907 | iterating over the 1M Vector after converting the Table with `makeVector()` and calculating the sums      |
+
+The memory impact is not very significant, amounting to 74.01MB for 1M dies compared with 44.65MB for the previously prototyped API.
+
+#### Previously proposed solutions
+
+These are the two possible solutions identified for representing the data in the memory. The fist one is an in-house solution:
 
```

Comment:
Unless I'm mistaken we can heavily edit the below section. First, we can rename 'Previously proposed solutions' to 'Alternative solutions'. Then, I would think the only solution we would need to call out is the in-house TypedArray API. The Apache Arrow API _is_ the API of choice, and thus doesn't need to be mentioned here. Finally you can remove the blurb at the bottom of the section that described how you will ultimately decide on the solution.